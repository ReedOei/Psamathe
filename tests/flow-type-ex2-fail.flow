contract Main {
}

contract Wrong {
}

;

var x : empty Main
new Wrong() --> x

