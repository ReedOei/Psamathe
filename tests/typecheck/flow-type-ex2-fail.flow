contract Main {
}

contract Wrong {
}

;

var x : ! Main := new Wrong()

