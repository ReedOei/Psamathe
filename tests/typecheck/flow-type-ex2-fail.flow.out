<generatedTop>
  <common>
    <k>
      new Wrong ( .Exprs ) : ! Main ~> insert ( x , ! Main ) ~> pass ~> drop ( x ) ~> .
    </k>
    <contracts>
      Main |-> contract Main { .Decls }
      Wrong |-> contract Wrong { .Decls }
    </contracts>
    <fields>
      Main |-> .Set
      Wrong |-> .Set
    </fields>
    <constructors>
      Main |-> constructor ( .VarDefs , pass )
      Wrong |-> constructor ( .VarDefs , pass )
    </constructors>
    <transactions>
      .Map
    </transactions>
    <types>
      address |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , address )
      bool |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , bool )
      nat |-> modified ( SetItem ( consumable )
      SetItem ( fungible ) , nat )
      string |-> modified ( SetItem ( consumable )
      SetItem ( nonfungible ) , string )
    </types>
    <freshCounter>
      0
    </freshCounter>
  </common>
  <typeEnv>
    .Map
  </typeEnv>
</generatedTop>
